static/js/silhouette.js

```diff
@@ -0,0 +1,142 @@
+'use strict'
+
+/**
+ * Calculate euclidean distance between two points
+ *
+ * @function euclideanDist
+ * @param {array} x - coordinate of point x
+ * @param {array} y - coordinate of point y
+ * @return {number} euclidean distance between x and y
+ */
+function euclideanDist(x, y) {
+  // check x, y
+  if (x == y) {
+    return 0;
+  }
+  let sum = 0;
+  for (let i = 0; i < x.length; i++) {
+    sum += (x[i] - y[i]) ** 2;
+  }
+  return Math.sqrt(sum);
+}
+
+/**
+ * return the pairwise distance matrix of each point in the input data array
+ *
+ * @function pairwiseDist
+ * @param {Array} x - the input data array
+ * @return {Array} distance matrix of the given data
+ */
+//TODO: add metrics='euclideanDist'
+function pairwiseDist(x) {
```

Comment:
I am thinking if we can just follow the Python names, since you already did so to `bincount`:
`pairwiseDist` => `pdist` (https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.pdist.html)
`euclideanDist` => `euclidean` (https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.euclidean.html)

And add a comment to the docstring: `...the function's behavior is analogus to scipy function `pdist`)

Reply:
function names updated. I am not sure about the exact way of naming as in sklearn:
https://github.com/scikit-learn/scikit-learn/blob/fd237278e895b42abe8d8d09105cbb82dc2cbba7/sklearn/metrics/pairwise.py#L201
https://github.com/scikit-learn/scikit-learn/blob/fd237278e895b42abe8d8d09105cbb82dc2cbba7/sklearn/metrics/pairwise.py#L1465

Reply:
@nujinuji Got it! Thanks for pointing me to the scikit-learn code. I think the naming scheme is up to you. I don't have a strong preference. Feel free to use names that are most convenient to you!